### PR TITLE
Add note about auditor to How-To-Open-a-Homebrew-Pull-Request.md

### PR DESCRIPTION
**Pull request by jrschumacher**
During my [submission](https://github.com/Homebrew/homebrew-core/pull/182146) I followed this guide since it is the first link, but it wasn't until we got to the end of the process that I learned I should have created a tap.

This was a great experience for me, but a waste of maintainers time. 

- [ ] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

### CODE REVIEW

File: docs/How-To-Open-a-Homebrew-Pull-Request.md [hunk 1]
```diff
@@ -16,6 +16,15 @@ The type of change you want to make influences which of Homebrew's main reposito
...
 
+Before getting started, please note that we cannot distribute every single piece of software and have to draw a line on “notability” somewhere. Of course, we would be happy to reconsider in the future. If you don't meet notability requirements you might consider [creating your own tap](https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap).
+
+Notability [auditor](https://github.com/Homebrew/brew/blob/638a3dc3a33c4c2029db058cf3d399008621e973/Library/Homebrew/utils/shared_audits.rb) will fail on repositories that do not meet:
+
+- **>29** days old
+- **>29** forks
+- **>29** watchers (Github only)
+- **>74** stars
+
```
Comment (MikeMcQuaid) [inline docs/How-To-Open-a-Homebrew-Pull-Request.md:27]: Hey @jrschumacher, thanks for the PR!

A link to the "Acceptable Formulae" page might be more suitable here. Also, this should be under "Formulae-related pull request". If you're feeling really nice: a link to "Acceptable Casks" under "Cask-related pull request" would be great, too!

Comment (jrschumacher) [reply docs/How-To-Open-a-Homebrew-Pull-Request.md:27]: Happy to help. I'll try and get it done today.

File: docs/How-To-Open-a-Homebrew-Pull-Request.md [hunk 2]
```diff
@@ -58,6 +67,15 @@ The type of change you want to make influences which of Homebrew's main reposito
...
 
+Before getting started with a new cask, please read the [Acceptable Casks](https://github.com/Homebrew/brew/blob/master/docs/Acceptable-Casks.md) and note that we cannot distribute every single piece of software and have to draw a line on “notability” somewhere. If you don't meet notability requirements you might consider [creating your own tap](https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap) (casks can also be installed from a tap). Of course, we would be happy to reconsider in the future!
+
+Notability [auditor](https://github.com/Homebrew/brew/blob/638a3dc3a33c4c2029db058cf3d399008621e973/Library/Homebrew/utils/shared_audits.rb) will fail on repositories that do not meet:
+
+* **>29** days old
+* **>29** forks
+* **>29** watchers (Github only)
+* **>74** stars
+
```
Comment (MikeMcQuaid) [inline docs/How-To-Open-a-Homebrew-Pull-Request.md:78]:
```suggestion
Before creating a new cask, please read [Acceptable Casks](https://github.com/Homebrew/brew/blob/master/docs/Acceptable-Casks.md).
```

**Review by MikeMcQuaid — COMMENTED** [review]
Don't think we need to have specific audit output in here, thanks!

File: docs/How-To-Open-a-Homebrew-Pull-Request.md [hunk 3]
```diff
@@ -34,6 +34,7 @@ The type of change you want to make influences which of Homebrew's main reposito
 
 ### Formulae-related pull request
 
+Before creating a new formulae, please read [Acceptable Formulae](https://github.com/Homebrew/brew/blob/master/docs/Acceptable-Formulae.md).
```
Comment (ZhongRuoyu) [inline docs/How-To-Open-a-Homebrew-Pull-Request.md:37]:
```suggestion
Before creating a new formulae, please read [Acceptable Formulae](https://docs.brew.sh/Acceptable-Formulae).
```

File: docs/How-To-Open-a-Homebrew-Pull-Request.md [hunk 4]
```diff
@@ -58,6 +59,7 @@ The type of change you want to make influences which of Homebrew's main reposito
 
 ### Cask-related pull request
 
+Before creating a new cask, please read [Acceptable Casks](https://github.com/Homebrew/brew/blob/master/docs/Acceptable-Casks.md).
```
Comment (ZhongRuoyu) [inline docs/How-To-Open-a-Homebrew-Pull-Request.md:62]:
```suggestion
Before creating a new cask, please read [Acceptable Casks](https://docs.brew.sh/Acceptable-Casks).
```

**Review by MikeMcQuaid — APPROVED** [review]
Thanks @jrschumacher!

**Review by ZhongRuoyu — APPROVED** [review]
Thanks @jrschumacher! I suggest using URLs from https://docs.brew.sh/ instead because that's where we expect people to read docs (rather than here in the repo).

File: docs/How-To-Open-a-Homebrew-Pull-Request.md (hunk 3 above)
Comment (MikeMcQuaid) [inline docs/How-To-Open-a-Homebrew-Pull-Request.md:37]:
```suggestion
Before creating a new formulae, please read [Acceptable Formulae](https://github.com/Homebrew/brew/blob/master/docs/Acceptable-Formulae.md).

```

File: docs/How-To-Open-a-Homebrew-Pull-Request.md (hunk 4 above)
Comment (MikeMcQuaid) [inline docs/How-To-Open-a-Homebrew-Pull-Request.md:62]:
```suggestion
Before creating a new cask, please read [Acceptable Casks](https://github.com/Homebrew/brew/blob/master/docs/Acceptable-Casks.md).

```

**Review by MikeMcQuaid — COMMENTED** [review]
Whoops, CI failure: https://github.com/Homebrew/brew/actions/runs/10616387548/job/29426655496?pr=18189

File: docs/How-To-Open-a-Homebrew-Pull-Request.md [hunk 5]
```diff
@@ -34,6 +34,9 @@ The type of change you want to make influences which of Homebrew's main reposito
 
 ### Formulae-related pull request
 
+Before creating a new formulae, please read [Acceptable Formulae](https://docs.brew.sh/Acceptable-Formulae).
+
+
```
Comment (carlocab) [inline docs/How-To-Open-a-Homebrew-Pull-Request.md:39]:
```suggestion
```

**Review by carlocab — APPROVED** [review]
Thanks for the PR!

One suggestion; this should fix the CI failure.

**Review by carlocab — APPROVED** [review]
Thank you!

**Review by ZhongRuoyu — APPROVED** [review]

**Pull request merged**